src/Sema.zig

```diff
@@ -32080,14 +32080,86 @@ fn analyzeSlice(
         .Pointer => switch (ptr_ptr_child_ty.ptrSize(mod)) {
             .One => {
                 const double_child_ty = ptr_ptr_child_ty.childType(mod);
+                ptr_or_slice = try sema.analyzeLoad(block, src, ptr_ptr, ptr_src);
                 if (double_child_ty.zigTypeTag(mod) == .Array) {
                     ptr_sentinel = double_child_ty.sentinel(mod);
-                    ptr_or_slice = try sema.analyzeLoad(block, src, ptr_ptr, ptr_src);
                     slice_ty = ptr_ptr_child_ty;
                     array_ty = double_child_ty;
                     elem_ty = double_child_ty.childType(mod);
                 } else {
-                    return sema.fail(block, src, "slice of single-item pointer", .{});
+                    const bounds_error_message = "slice of single-item pointer must have comptime-known bounds of 0 and 1";
```

Comment:
This doesn't include that `0..0` is allowed since I didn't realize it would be, how about:
```suggestion
                    const bounds_error_message = "slice of single-item pointer must be '0..0' or '0.1'";
```

Reply:
Should I infer from this suggestion that you think I should special case `ptr[1..1]` to disallow it?

Reply:
Seems to me like that should be "out of bounds".

Reply:
It would be out-of-bounds, and hence UB to deference the pointer, but it is allowed to slice a `*[1]T` or `[1]T` as `[1..1]`; this file:
```zig
test {
    var byte: u8 = 1;

    const ptr = &byte;
    const array_ptr: *[1]u8 = ptr;
    _ = array_ptr[0..0];
    _ = array_ptr[0..1];
    _ = array_ptr[1..1];
    _ = array_ptr[2..2];
}
```
only has an error on the `_ = array_ptr[2..2];` line. That's why I'm asking - if we want the semantics to be like there was a coercion `*T -> *[1]T` before slicing then we should allow `[1..1]`.

Reply:
> but it is allowed to slice a `*[1]T` or `[1]T` as `[1..1]`; this file:

I didn't realize that was possible, I guess it should be allowed then.

Reply:
I've updated the error message to say it must have `comptime-known bounds [0..0], [0..1], or [1..1]`.